### PR TITLE
Intrinsify the atexit function, Closes #264

### LINKED ISSUE #264: Intrinsify the atexit C library function
The [C library function `atexit`](http://www.tutorialspoint.com/c_standard_library/c_function_atexit.htm) allows a programmer to register a function that is called before a program terminates. We need to intrinsify this function in Sulong since we cannot simply pass a Truffle function to the native side (see #57). The implementation would be conceptually similar to the constructor and destructor attributes (see #214). In the LLVM test suite there is a test case `test-suite-3.2.src/SingleSource/UnitTests/2003-05-14-AtExit.c` that could validate the implementation.

### PR DESCRIPTION
**Pull request by pointhi**
Closes: #264

http://www.cplusplus.com/reference/cstdlib/atexit/?kw=atexit

### CODE REVIEW

File: projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import com.oracle.truffle.llvm.nodes.impl.func.LLVMFunctionStartNode;
+import com.oracle.truffle.llvm.nodes.impl.func.LLVMCallNode.LLVMUnresolvedCallNode;
+import com.oracle.truffle.llvm.nodes.impl.intrinsics.llvm.LLVMIntrinsic.LLVMI32Intrinsic;
+import com.oracle.truffle.llvm.nodes.impl.literals.LLVMFunctionLiteralNodeGen;
+import com.oracle.truffle.llvm.nodes.impl.literals.LLVMSimpleLiteralNode.LLVMAddressLiteralNode;
+import com.oracle.truffle.llvm.types.LLVMFunctionDescriptor;
+import com.oracle.truffle.llvm.types.LLVMFunctionDescriptor.LLVMRuntimeType;
+
+@NodeChild(type = LLVMFunctionNode.class, value = "func")
+public abstract class LLVMAtExit extends LLVMI32Intrinsic {
```
Comment (jkreindl) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java]: I am not sure if there is a convention for that, but shouldn't this class have Node or Intrinsic at the end of its name?

File: projects/com.oracle.truffle.llvm.test/tests/c/atexit/atexit003.c [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+static void exit1() { printf("something went wrong!\n"); }
```
Comment (jkreindl) [inline projects/com.oracle.truffle.llvm.test/tests/c/atexit/atexit003.c]: Could you change this to be more descriptive like 'error: functions registered using atexit should only be executed after successful program execution' or something of the like?

File: projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java (hunk 1 above)
Comment (pointhi) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java]: at least for c intrinsic, this nomenclature isn't used yet (with two exceptions)

File: projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        LLVMFunctionNode functionNode = LLVMFunctionLiteralNodeGen.create(func);
+
+        LLVMUnresolvedCallNode callNode = new LLVMUnresolvedCallNode(functionNode, args, LLVMRuntimeType.VOID, context);
+
+        RootCallTarget callTarget = Truffle.getRuntime().createCallTarget(
+                        new LLVMFunctionStartNode(callNode,
+                                        new LLVMNode[]{},
+                                        new LLVMNode[]{},
+                                        null,
+                                        new FrameDescriptor(), "", new LLVMStackFrameNuller[0]));
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java]: You can pass `func.getName()` to have a useful name for the `callTarget`.

File: projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java (hunk 1 above)
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMAtExit.java]: Yes, we have that exception for intrinsics and messages. @jkreindl if you want to change this you can open an issue to start a discussion.

File: projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/func/LLVMGlobalRootNode.java [hunk 4]
```diff
@@ -109,6 +110,8 @@ public Object execute(VirtualFrame frame) {
         }
     }
 
+    private static final int UNIX_SIGABORT = 134;
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/func/LLVMGlobalRootNode.java]: This constant is already defined in `LLVMAbort`.

**Pull request merged**